File: exercises/exercise-1-testing-java/src/main/java/io/miragon/example/adapter/outbound/zeebe/bike/BikeSubscriptionProcessAdapter.java
package io.miragon.example.adapter.outbound.zeebe.bike;

import io.miragon.example.adapter.process.config.ProcessEngineApi;
import io.miragon.example.adapter.process.generated.BikeSubscriptionSignupProcessApi;
import io.miragon.example.application.port.outbound.bike.BikeSubscriptionProcess;
import io.miragon.example.domain.bike.BikeId;
import io.miragon.example.domain.bike.BikeSubscriptionId;
import org.springframework.stereotype.Component;

import java.util.Map;

@Component
public class BikeSubscriptionProcessAdapter implements BikeSubscriptionProcess {

    private final ProcessEngineApi engineApi;

    public BikeSubscriptionProcessAdapter(ProcessEngineApi engineApi) {
        this.engineApi = engineApi;
    }

    @Override
    public long startSubscription(BikeSubscriptionId id, BikeId bikeId) {
        Map<String, Object> variables = Map.of(
            "subscriptionId", id.value().toString(),
            "bikeId", bikeId.value().toString()
        );
        return engineApi.startProcess(
            BikeSubscriptionSignupProcessApi.PROCESS_ID,
            variables
        );
    }

    @Override
    public void sendPaymentReceived(BikeSubscriptionId id) {
        engineApi.sendMessage(
            BikeSubscriptionSignupProcessApi.Messages.MESSAGE_PAYMENT_RECEIVED,
            id.value().toString()
        );
    }

    @Override
    public void sendRequestCanceled(BikeSubscriptionId id) {
        engineApi.sendMessage(
            BikeSubscriptionSignupProcessApi.Messages.MESSAGE_REQUEST_CANCELED,
            id.value().toString()
        );
    }

    @Override
    public void sendBikeReceived(BikeSubscriptionId id) {
        engineApi.sendMessage(
            BikeSubscriptionSignupProcessApi.Messages.MESSAGE_BIKE_RECEIVED,
            id.value().toString()
        );
    }

}

File: exercises/exercise-1-testing-kotlin/src/main/kotlin/io/miragon/example/adapter/outbound/zeebe/bike/BikeSubscriptionProcessAdapter.kt
package io.miragon.example.adapter.outbound.zeebe.bike

import io.miragon.example.adapter.process.generated.BikeSubscriptionSignupProcessApi
import io.miragon.example.application.port.outbound.bike.BikeSubscriptionProcess
import io.miragon.example.adapter.process.config.ProcessEngineApi
import io.miragon.example.domain.bike.BikeId
import io.miragon.example.domain.bike.BikeSubscriptionId
import org.springframework.stereotype.Component

@Component
class BikeSubscriptionProcessAdapter(
    private val engineApi: ProcessEngineApi
) : BikeSubscriptionProcess {

    override fun startSubscription(id: BikeSubscriptionId, bikeId: BikeId): Long {
        val variables = mapOf(
            "subscriptionId" to id.value.toString(),
            "bikeId" to bikeId.value.toString()
        )
        return engineApi.startProcess(
            processId = BikeSubscriptionSignupProcessApi.PROCESS_ID,
            variables = variables
        )
    }

    override fun sendPaymentReceived(id: BikeSubscriptionId) {
        engineApi.sendMessage(
            messageName = BikeSubscriptionSignupProcessApi.Messages.MESSAGE_PAYMENT_RECEIVED,
            correlationId = id.value.toString()
        )
    }

    override fun sendRequestCanceled(id: BikeSubscriptionId) {
        engineApi.sendMessage(
            messageName = BikeSubscriptionSignupProcessApi.Messages.MESSAGE_REQUEST_CANCELED,
            correlationId = id.value.toString()
        )
    }

    override fun sendBikeReceived(id: BikeSubscriptionId) {
        engineApi.sendMessage(
            messageName = BikeSubscriptionSignupProcessApi.Messages.MESSAGE_BIKE_RECEIVED,
            correlationId = id.value.toString()
        )
    }
}
